core/dbt/clients/system.py

```diff
@@ -478,6 +478,24 @@ def untar_package(tar_path: str, dest_dir: str, rename_to: Optional[str] = None)
         dbt.clients.system.rename(downloaded_path, desired_path, force=True)
 
 
+def resolve_tar_dir_name(TarFile: tarfile.TarFile):
+    tarfile_dir_list = [x.name for x in TarFile.getmembers() if x.isdir()]
+    tar_dir_name = os.path.commonpath(tarfile_dir_list)
+    return convert_path(tar_dir_name)
+
+
+def untar_tarfile(
+    TarFile: tarfile.TarFile, dest_dir: str, rename_to: Optional[str] = None
+) -> None:
+    tar_dir_name = dbt.clients.system.resolve_tar_dir_name(TarFile)
```

Comment:
You can update `untar_package` to use `resolve_tar_dir_name` which will remove the need for the new `untar_tarfile`.   You would just pass through the tar path instead of the TarFile object.

Reply:
Based on your feedback I've removed everything I initially wrote around passing the TarFile object around, in preference for using a tar path. I've removed this `untar_tarfile` function, and moved over to untar_package() as you suggest.